### PR TITLE
Added logic in iAssertion.looksLike() to handle HttpResponse objects.

### PR DESCRIPTION
**Pull request by ncalaway**
Having the body of every HttpResponse was corrupting images I tried to download. I don't know if anything will break by not forcing every HttpResponse.body to be a string, but this certainly fixes downloading images. Text downloaded is still represented as a string.

### CODE REVIEW

File: src/value.ts [hunk 1]
```diff
@@ -764,10 +764,15 @@ export class Value implements iValue {
       ...opts,
     });
     const resp = await request.fetch();
+    let file: string | Buffer = resp.body;
     if (localFilePath) {
-      fs.writeFileSync(localFilePath, resp.body);
+      if (resp.trailers["content-type"].split("/")[0] === "image") {
```
Comment (jasonbyrne) [inline src/value.ts:769]: This seems out of place here inside the `if (localFilePath) {` feels like it should be part of line 767 instead.

I'm also uncertain about changing the return type here from HttpResponse to the response body. Something might be relying on the rest of the response

File: src/httpresponse.ts [hunk 2]
```diff
@@ -43,18 +44,23 @@ export class HttpResponse {
   }
 
   static fromNeedle(response: NeedleResponse): HttpResponse {
+    const contentType = response.headers["content-type"];
+    response.trailers["content-type"] = contentType;
```
Comment (jasonbyrne) [inline src/httpresponse.ts:48]: I'm curious why we need to use trailers here instead of the header.

File: tests/src/visual.ts [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
   .next("Logo", async (context) => {
     // ✕  Logo should look like the control.
     //  …  Actual: Error: Input image is invalid.
-    // const logo = await context.exists("img[alt='Google']");
-    // context
-    //   .assert("Logo should look like the control.", await logo.download())
-    //   .looksLike("@google");
+    const logo = await context.exists("img[alt=Debian]");
+    const download = await logo.download(
+      "/Users/nick/Documents/GitHub/flagpole/tests/images/debian.png"
```
Comment (jasonbyrne) [inline tests/src/visual.ts:14]: We should make this a relative path that is not absolute to your machine

**Review by jasonbyrne — CHANGES_REQUESTED** [review]

File: src/httpresponse.ts [hunk 4]
```diff
@@ -43,18 +44,22 @@ export class HttpResponse {
...
 
   static fromNeedle(response: NeedleResponse): HttpResponse {
+    const contentType = response.headers["content-type"];
     const r = new HttpResponse({
       status: [response.statusCode || 0, response.statusMessage || ""],
       headers: <KeyValue>response.headers,
       body:
         typeof response.body === "string"
           ? response.body
+          : contentType?.split("/")[0] === "image"
```
Comment (jasonbyrne) [inline src/httpresponse.ts:54]: this would be more readable as `contentType.startsWith('image')`

File: src/value.ts [hunk 5]
```diff
@@ -764,10 +764,15 @@ export class Value implements iValue {
       ...opts,
     });
     const resp = await request.fetch();
+    let file: string | Buffer = resp.body;
+    if (resp.headers["content-type"].split("/")[0] === "image") {
+      file = Buffer.from(resp.body, "base64");
+    }
```
Comment (jasonbyrne) [inline src/value.ts:770]: I don't like setting something and then immediately resetting it if we don't have to do that. 

```
const file: string | Buffer = resp.headers["content-type"].startsWith("image") ?  
   Buffer.from(resp.body, "base64") : resp.body;
```

would be cleaner

File: src/value.ts [hunk 6]
```diff
@@ -764,10 +764,15 @@ export class Value implements iValue {
...
     if (localFilePath) {
-      fs.writeFileSync(localFilePath, resp.body);
+      fs.writeFileSync(localFilePath, file);
     }
-    return resp;
+
+    return file;
```
Comment (jasonbyrne) [inline src/value.ts:775]: I'm still confused why we are returning the file contents here instead of the response like before

**Review by jasonbyrne — CHANGES_REQUESTED** [review]

**Comment by ncalaway** [comment]
@jasonbyrne 
> I'm still confused why we are returning the file contents here instead of the response like before

I'm doing this because of the testing code here:
```suite
  .html("Cheerio test of Google Logo")
  .open("/")
  .next("Logo", async (context) => {
    // ✕  Logo should look like the control.
    //  …  Actual: Error: Input image is invalid.
    // const logo = await context.exists("img[alt='Google']");
    // context
    //   .assert("Logo should look like the control.", await logo.download())
    //   .looksLike("@google");
  });
```

It's trying to do a looksLike assertion on an HttpResponse, which wouldn't work.

We could return the HttpResponse and call the assertion on HttpResponse.body, but that would not work with images because it has to be a string. If we could allow HttpResponse.body to be a string and a buffer, we could keep returning the HttpResponse. I probably wouldn't even have to bother with the base64 encoding/decoding if we did that.

**Comment by ncalaway** [comment]
@jasonbyrne per our discussion on Slack, I restored iValue.download() to its previous functionality and added logic for iAssertion.looksLike() to handle HttpResponse objects.

**Review by jasonbyrne — APPROVED** [review]

**Pull request merged**